src/v/kafka/server/quota_manager.cc

```diff
@@ -151,6 +155,69 @@ quota_manager::add_quota_id(std::string_view qid, clock::time_point now) {
     co_await _client_quotas.update(std::move(update_func));
 }
 
+void quota_manager::update_client_quotas() {
+    vassert(
+      ss::this_shard_id() == _client_quotas.shard_id(),
+      "update_client_quotas must only be called on the owner shard");
+
+    ssx::spawn_with_gate(_gate, [this] {
+        return _client_quotas.update([this](client_quotas_map_t quotas) {
```

Comment:
question: we need to make this change inside a `sharded_ptr::update` only to prevent any concurrent inserts/deletes to the map, right?

Reply:
> question: we need to make this change inside a `sharded_ptr::update` only to prevent any concurrent inserts/deletes to the map, right?

Is that not what it's doing?